File: prova-c/1043/gabarito.c
#include <stdio.h>

double max(double a, double b)
{
    return (a > b) ? a : b;
}

void main()
{
    double a, b, c;
    scanf("%lf %lf %lf", &a, &b, &c);
    double perimetro = a + b + c,
           maior = max(a, max(b, c));

    if (perimetro <= 2 * maior)
        printf("Area = %.1lf\n", (a + b) * c / 2);
    else
        printf("Perimetro = %1f\n", perimetro);
}

File: prova-c/1043/gabarito.py
nums = [*map(float, input().split())]
total = sum(nums)

if total <= 2 * max(nums):
    a, b, c = nums
    area = (a + b) * c / 2
    print(f"Area = {area:.1f}")
else:
    print(f"Perimetro = {total:.1f}")
